src/main/java/net/minecraftforge/common/Dimension.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common;
+
+import java.util.Map;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.DimensionType;
+import net.minecraft.world.WorldProvider;
+
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import net.minecraftforge.registries.ForgeRegistry;
+import net.minecraftforge.registries.IForgeRegistry;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+/**
+ * Should be created and registered at {@link net.minecraftforge.event.RegistryEvent.Register<Dimension>}
+ * If the number of dimensions you need it variable it's probably best to register them at the start of the game and initialize them later
+ * @author temp1011 (needed?)
+ */
+public class Dimension implements IForgeRegistryEntry<Dimension>
```

Comment:
This should have a different name; 1.13 uses this name, and I can see these being mixed up.

Reply:
Yeah I'm trying to think of a new one. `WorldInstance`? `DimensionInfo`? `DimensionProvider`? Not sure at the moment, I don't have anything particularly good.

Reply:
DimensionProvider sound not too bad.